### ISSUE TITLE
what's the status of the library?

### ISSUE THREAD

**Issue by kulak** [pixelbender/go-dtls#3]
I am interested in adding DTLS support to https://github.com/dustin/go-coap library and yours seems to be the best bet, unless I want to deal with openssl.

So, I wounder if I can attempt integration or it is just too rough at the moment.

Thank you

**Comment by pixelbender** [comment]
@Kulak In developing, and it's not yet ready for use. Now I'm implementing internal/tls package with pluggable transport (UDP for DTLS with retransmission and ordering stuff, or TCP for TLS). It makes testing easy.

**Comment by maufl** [comment]
Does that mean you're still intending to finish this library? I started my own [Go DTLS library](http://github.com/maufl/dtls) last year because I wanted to have one. However, I'm no (D)TLS expert and was struggling to get anything working. Some things are already working but I'm stuck at testing handshake message retransmission (this seems to me to be the most difficult part), because I don't know how to properly test it. I'm currently not motivated to continue working on it, but if I can help you finish your library let me know. I would be very happy to have a working Go DTLS implementation.

**Comment by pixelbender** [comment]
@maufl Yes. That's why I decided to implement TLS first. So, next week I plan to commit my work so you can see and decide what part you can improve. Thanks

**Comment by ghost** [comment]
Hey @pixelbender 

There is a group implementing lots of the same protocols you are and there is allot of momentum if your interested in this stuff.  They could do with help from people that know about this

https://github.com/pions/webrtc/issues/20

https://github.com/gortc

**Comment by hhuccpj** [comment]
Dear Sir, I would like to integrate your code to my CoAP project. Is it completed please? Seems no update for a long while.
